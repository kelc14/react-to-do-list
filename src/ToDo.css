.ToDo {
  display: flex;
  width: 100%;
  font-family: "Roboto Slab", serif;
  font-weight: 300;
  margin: 3px;
  padding: 7px;
}

.ToDo:hover {
  box-shadow: rgba(100, 100, 111, 0.2) 0px 7px 29px 0px;
  background-color: red;
  color: white;
}
