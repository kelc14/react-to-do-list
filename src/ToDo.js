import React from "react";
import "./ToDo.css";

const ToDo = ({ name, removeItem, id }) => {
  const handleRemove = (e) => {
    let itemId = e.target.id;
    console.log(itemId);
    removeItem(itemId);
  };
  return (
    <div className="ToDo" id={id} onClick={handleRemove}>
      {name}
    </div>
  );
};

export default ToDo;
