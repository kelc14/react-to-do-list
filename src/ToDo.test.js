import { render, fireEvent } from "@testing-library/react";
import ToDo from "./ToDo";

// smoke test
it("renders without crashing", function () {
  render(<ToDo name="Test To Do" id={1} />);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(<ToDo name="Test To Do" id={1} />);
  expect(asFragment()).toMatchSnapshot();
});
