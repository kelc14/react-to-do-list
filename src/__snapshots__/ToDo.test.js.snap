// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ToDo"
    id="1"
  >
    Test To Do
  </div>
</DocumentFragment>
`;
